datastreamer/streamserver.go

```diff
@@ -1102,6 +1203,65 @@ func (s *StreamServer) processCmdBookmark(client *client) error {
 	return nil
 }
 
+// processCmdLatestL2Block processes the TCP LatestL2Block command from the clients
+func (s *StreamServer) processCmdLatestL2Block(client *client) error {
+	// Log
+	log.Debugf("Client %s command LatestL2Block", client.clientID)
+
+	// Send a command result entry OK
+	err := s.sendResultEntry(0, "OK", client)
+	if err != nil {
+		return err
+	}
+
+	// Get the latest L2Block entry by searching backwards
+	entry, err := s.getLatestL2BlockEntry()
+	if err != nil {
+		log.Warnf("Latest L2Block entry not found: %v", err)
+		entry = FileEntry{}
+		entry.Length = FixedSizeFileEntry
+		entry.Type = EntryTypeNotFound
+	}
+	entry.packetType = PtDataRsp
+	binaryEntry := encodeFileEntryToBinary(entry)
+
+	// Send entry to the client
+	if client.conn != nil {
+		_, err = TimeoutWrite(client, binaryEntry, s.writeTimeout)
+	} else {
+		err = ErrNilConnection
+	}
+	if err != nil {
+		log.Errorf("Error sending latest L2Block entry to %s: %v", client.clientID, err)
+		return err
+	}
+
+	return nil
+}
+
+// getLatestL2BlockEntry returns the most recent L2Block entry using cached data for optimal performance
+// This function now only uses the cache and does not perform any file searches
+func (s *StreamServer) getLatestL2BlockEntry() (FileEntry, error) {
+	// PERFORMANCE OPTIMIZATION: Use cached L2Block entry only
+	s.latestL2BlockMutex.RLock()
+	cachedEntry := s.latestL2BlockEntry
```

Comment:
Since `initializeL2BlockCache` may falied and the error it returned is ignored, `s.latestL2BlockEntry` could be an empty value, which is unexpected.

Reply:
1. There a check below
`if cachedEntry.Type == internalEntryTypeL2Block && **cachedEntry.Length > 0*`
2. the RPC node also check if get the value ==0, will return error tool
https://github.com/okx/xlayer-erigon/pull/799/files#diff-477089124b274dd75c833b38e783d4cadf9ccaeb6c923e978c2cb092b69a846dR219